packages/cache/src/LocalCacheStorage.ts

```diff
@@ -31,13 +30,25 @@ export class LocalCacheStorage extends CacheStorage {
     });
 
     await Promise.all(
-      files.map(async file => {
-        await fs.mkdirp(path.dirname(path.join(this.cwd, file)));
-        await fs.copy(
-          path.join(localCacheFolder, file),
-          path.join(this.cwd, file)
-        );
-      })
+      files
+        .filter(async file => {
+          const src = path.join(localCacheFolder, file);
+          const dest = path.join(this.cwd, file);
+
+          if (fs.existsSync(dest)) {
+            const stats = await Promise.all([fs.stat(src), fs.stat(dest)]);
```

Comment:
existsSync isn't deprecated. But I get your point here. I can do a dest stat first. And do a src stat IF needed.

Reply:
Btw, to get rid of existSync, you have to dabble in the world of try-await-catch which itself can be slow...

Reply:
In C# it is not recommended to use try/catch because it is slow. In the contrary, in python it is recommended to beg forgiveness rather than asking permission. I am curious to learn about the recommendations for NodeJS, because I also found it weird that they recommend replacing `exists` by `stat`.